Mention that aliases are recursive

har7an opened the pull request:
Instead of duplicating portions of commands that are used identically across many aliases, the user can instead reuse any previously defined aliases.

<!-- homu-ignore:start -->
<!--
NOTICE: Due to limited review capacity, the Cargo team is not accepting new
features or major changes at this time. Please consult with the team before
opening a new PR. Only issues that have been explicitly marked as accepted
will be reviewed.

Thanks for submitting a pull request 🎉! Here are some tips for you:

* If this is your first contribution, read "Cargo Contribution Guide":
  https://doc.crates.io/contrib/
* Run `cargo fmt --all` to format your code changes.
* Small commits and pull requests are always preferable and easy to review.
* If your idea is large and needs feedback from the community, read how:
  https://doc.crates.io/contrib/process/#working-on-large-features
* Cargo takes care of compatibility. Read our design principles:
  https://doc.crates.io/contrib/design.html
* When changing help text of cargo commands, follow the steps to generate docs:
  https://github.com/rust-lang/cargo/tree/master/src/doc#building-the-man-pages
* If your PR is not finished, set it as "draft" PR or add "WIP" in its title.
* It's ok to use the CI resources to test your PR, but please don't abuse them.

### What does this PR try to resolve?

Explain the motivation behind this change.
A clear overview along with an in-depth explanation are helpful.

You can use `Fixes #<issue number>` to associate this PR to an existing issue.

Today I started working with [`xtask`](https://github.com/matklad/cargo-xtask/), which is a build system based on a simple cargo alias:

```toml
[alias]
xtask = "run --package xtask --"
```

Since the word "xtask" is rather difficult to type in my opinion (at least my left hand struggles quite a bit) I wanted to add another alias, `x` as a shorthand (similar to what `build`, `run`, etc. have by default). Thereby I discovered that I needn't replicate the whole alias, because aliases are recursive. I consulted the docs and couldn't find a mention of this, hence I'm adding it as part of this PR so other users can discover it.

### How should we test and review this PR?

Demonstrate how you test this change and guide reviewers through your PR.
With a smooth review process, a pull request usually gets reviewed quicker.

If you don't know how to write and run your tests, please read the guide:
https://doc.crates.io/contrib/tests

I don't think this requires a separate test, it's a minor change to the documentation only.

### Additional information

Other information you want to mention in this PR, such as prior arts,
future extensions, an unresolved problem, or a TODO list.
-->
<!-- homu-ignore:end -->

### What does this PR try to resolve?

Today I started working with [`xtask`](https://github.com/matklad/cargo-xtask/), which is a build system based on a simple cargo alias:

```toml
[alias]
xtask = "run --package xtask --"
```

Since the word "xtask" is rather difficult to type in my opinion (at least my left hand struggles quite a bit) I wanted to add another alias, `x` as a shorthand (similar to what `build`, `run`, etc. have by default). Thereby I discovered that I needn't replicate the whole alias, because aliases are recursive. I consulted the docs and couldn't find a mention of this, hence I'm adding it as part of this PR so other users can discover it.


### How should we test and review this PR?

I don't think this requires a separate test, it's a minor change to the documentation only.


### Additional information

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @ehuss (or someone else) soon.

Please see [the contribution instructions](https://github.com/rust-lang/cargo/blob/master/CONTRIBUTING.md) for more information.

epage commented on src/doc/src/reference/config.md:
````diff
@@ -335,6 +335,13 @@ r = "run"
 
+Aliases are recursive:
+
+```toml
+rr = "run --release"
+space_example = "rr -- \"command list\""
+```
````
Could you provide a more real-world example for `space_example`?

har7an replied:
@epage Sure, I just took the one from the `config.toml` shown in the docs. Would something like this qualify?

```
toml
rr = "run --release"
space_example = "rr --example example_from_space"
```

epage replied:
Ah, if its pre-existing, then thats fine.

How about we also update that earlier config to be recursive?  That way we people who just see that don't read the wrong things into it

har7an commented:
@epage I changed the example a bit:

```toml
[alias]
rr = "run --release"
recursive_example = "rr --example recursions"
```

And modified the `config.toml` further up on that page from this:

```toml
rr = "run --release"
space_example = ["run", "--release", "--", "\"command list\""]
```

to that:

```toml
rr = "run --release"
recursive_example = "rr --example recursions"
space_example = ["run", "--release", "--", "\"command list\""]
```

I preferred adding it as new example so we don't lose the one with the array notation. I also decided not to do the recursion in the array example so people don't get confused when skimming it. Is that alright?

epage commented:
@bors r+

bors commented:
:pushpin: Commit e8cd6f4bcfad96b1a81259f4d55584b6fa989610 has been approved by `epage`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/cargo) for this repository.

<!-- @bors r=epage e8cd6f4bcfad96b1a81259f4d55584b6fa989610 -->
<!-- homu: {"type":"Approved","sha":"e8cd6f4bcfad96b1a81259f4d55584b6fa989610","approver":"epage","queue":"https://bors.rust-lang.org/homu/queue/cargo"} -->

bors commented:
:hourglass: Testing commit e8cd6f4bcfad96b1a81259f4d55584b6fa989610 with merge 5ac9be2eeac607c66a4f7f962ad5934d2bfa54de...
<!-- homu: {"type":"BuildStarted","head_sha":"e8cd6f4bcfad96b1a81259f4d55584b6fa989610","merge_sha":"5ac9be2eeac607c66a4f7f962ad5934d2bfa54de"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang/cargo/runs/7674988551?check_suite_focus=true)
Approved by: epage
Pushing 5ac9be2eeac607c66a4f7f962ad5934d2bfa54de to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"epage","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang/cargo/runs/7674988551?check_suite_focus=true"},"merge_sha":"5ac9be2eeac607c66a4f7f962ad5934d2bfa54de"} -->
